EIPS/eip-7702.md

```diff
@@ -23,63 +23,126 @@ There is a lot of interest in adding short-term functionality improvements to EO
+The fields `chain_id`, `nonce`, `max_priority_fee_per_gas`, `max_fee_per_gas`, `gas_limit`, `destination`, `value`, `data`, and `access_list` follow the same semantics as [EIP-1559](./eip-1559.md).
+
+The `authorization_list` is a list of tuples that store the address to code which the signer desires to set in their EOA temporarily.
 
-At the start of executing the transaction, for each `[contract_code, y_parity, r, s]` tuple:
+The [EIP-2718](./eip-2718.md) `ReceiptPayload` for this transaction is `rlp([status, cumulative_transaction_gas_used, logs_bloom, logs])`.
 
-1. Let `signer = ecrecover(keccak(MAGIC + contract_code), y_parity, r, s]`.
-2. Verify that the contract code of `signer` is empty.
-3. Set the contract code of `signer` to `contract_code`.
-4. Add the `signer` account to `accessed_addresses` (as defined in [EIP-2929](./eip-2929.md).)
+#### Behavior
 
-At the end of the transaction, set the `contract_code` of each `signer` back to empty.
+At the start of executing the transaction, for each `[address, y_parity, r, s]` tuple:
 
-Note that the signer of any of the `contract_code` signatures, and the `tx.origin` of the transaction, are allowed to be different.
+1. Let `authority = ecrecover(keccak(MAGIC || address), y_parity, r, s]`.
+2. Verify that the code of `authority` is empty.
+3. Set the code of `authority` to code associated with `address`.
+4. Add the `authority` account to `accessed_addresses` (as defined in [EIP-2929](./eip-2929.md).)
+
+At the end of the transaction, set the code of each `authority` back to empty.
+
+Note that the signer of an authorization tuple may be different than `tx.origin` of the transaction.
+
+#### Gas Costs
+
+The intrinsic cost of the new transaction is inherited from [EIP-2930](./eip-2930.md), specifically `21000 + 16 * non-zero calldata bytes + 4 * zero calldata bytes + 1900 * access list storage key count + 2400 * access list address count`. Additionally, we add a cost of `PER_CONTRACT_CODE_BASE_COST * authorization list length`.
 
 ## Rationale
 
-### Conversion of EIP-3074 use cases
+### No initcode
+
+Running initcode is not desirable for many reasons. The chief concern is it's unnatural. Initcode is intended to initialize and deploy a contracts. With this EIP, it will take on a new role of determine whether it is appropriate to deploy code to the EOA. Suppose a user only wants code deployed to their account if they also have an operation bundled with the general transaction calldata. This gives EOA a unique power to control when and what code executes in their account. Although [EIP-7702](./eip-7702.md) as written still allows this to a degree, the lack of programmability in the decision will force wallets to not sign many authorization tuples and instead focus signing only a tuple pointing to a configurable proxy. This affords EOAs a similar experience to smart contract wallets
```

Comment:
```suggestion
Running initcode is not desirable for many reasons. The chief concern is it's unnatural. Initcode is intended to initialize and deploy contracts. With this EIP, it will take on a new role of determine whether it is appropriate to deploy code to the EOA. Suppose a user only wants code deployed to their account if they also have an operation bundled with the general transaction calldata. This gives EOA a unique power to control when and what code executes in their account. Although [EIP-7702](./eip-7702.md) as written still allows this to a degree, the lack of programmability in the decision will force wallets to not sign many authorization tuples and instead focus signing only a tuple pointing to a configurable proxy. This affords EOAs a similar experience to smart contract wallets
```